### PR TITLE
fix(ai-help): reset user quota on openai api error

### PR DESCRIPTION
**Pull request by argl**
If there is an error talking to the OpenAI API, we reinstate the user's quota.
(MP-708)

### CODE REVIEW

File: src/db/ai_help.rs [hunk 1]
```diff
@@ -114,6 +114,24 @@ pub fn create_or_increment_limit(
...
+            .execute(conn)?;
+    } else {
+        update(limits::table)
+            .filter(limits::user_id.eq(&user.id))
+            .set((
+                (limits::total_questions.eq(limits::total_questions - 1)),
+                (limits::session_questions.eq(limits::session_questions - 1)),
+            ))
+            .execute(conn)?;
+    }
```
Comment (caugner) [inline src/db/ai_help.rs:131]: Afaik we increment `session_question` and `total_questions` for both subscribers and free users, so we should also decrement both. No?

```suggestion
    update(limits::table)
        .filter(limits::user_id.eq(&user.id))
        .set((
            (limits::total_questions.eq(limits::total_questions - 1)),
            (limits::session_questions.eq(limits::session_questions - 1)),
        ))
        .execute(conn)?;
```

**Review by caugner — CHANGES_REQUESTED** [review]
Ovrerall LGTM, just one comment regarding different behavior for subscribers.

File: src/db/ai_help.rs (hunk 1 above)
Comment (argl) [reply src/db/ai_help.rs:131]: Sure, I was confused by the original code using `create_or_increment_total` and `create_or_increment_limit` for the different subscriber types, but these are all around actually limiting. Our aim here is not to enforce the limit but set the numbers right.

**Review by caugner — APPROVED** [review]
LGTM, thanks!

**Pull request merged**